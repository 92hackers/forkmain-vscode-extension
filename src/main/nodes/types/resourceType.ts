import { IK8sResource } from "../../domain/IK8sResource";

export interface List {
  apiVersion: string;
  items: Array<IK8sResource>;
  kind: string;
}

export interface PodResource extends IK8sResource {
  spec: {
    containers: Array<{
      name: string;
    }>;
  };
}
