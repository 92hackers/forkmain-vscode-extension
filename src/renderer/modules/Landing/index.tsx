import React from "react";

const Landing: React.FC = () => {
  return <div data-testid="landing"></div>;
};

export default Landing;
