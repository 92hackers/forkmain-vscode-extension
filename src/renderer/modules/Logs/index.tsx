import React, { useContext, useEffect, useRef } from "react";
import { useHistory } from "react-router-dom";
import { createStyles, makeStyles } from "@material-ui/core";
import qs from "qs";
import hljs from "highlight.js";
import { store } from "../../store/store";
import { CustomThemeOptions } from "../../themes";
import { ThemeType, LOG_INTERVAL_MS, LOG_TAIL_COUNT } from "../../constants";
import fetchLogs from "../../services/fetchLogs";

const useStyles = makeStyles((theme: CustomThemeOptions) =>
  createStyles({
    root: {
      width: "100%",
      height: "100%",
      overflowX: "hidden",
      overflowY: "auto",
      "&:hover": {
        "& div.logs-anthor": {
          display: "none",
        },
      },
    },
    noContent: {
      width: "100%",
      height: "100%",
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
    },
    list: {
      padding: 10,
      fontFamily: "droidsansmono",
      fontSize: 12,
    },
    line: {
      "&:before": {
        display: "inline-block",
        content: "attr(data-line)",
        marginRight: 12,
        color: theme.palette?.text?.hint,
        minWidth: 25,
        textAlign: "right",
      },
    },
  })
);

const Logs: React.FC = () => {
  const {
    state: { logs, theme },
  } = useContext(store);
  const history = useHistory();
  const search: string = history.location.search;
  const elementRef: React.MutableRefObject<HTMLDivElement | null> = useRef(
    null
  );
  const classes = useStyles();

  const renderNoContent = () => (
    <div className={classes.noContent}>No Content.</div>
  );
  const renderList = () => {
    return (
      <>
        <ul className={classes.list}>
          {logs.items.map((text: string, i: number) => {
            hljs.configure({});
            const result = hljs.highlight("haskell", text);
            return (
              <li
                key={i}
                data-line={i + 1}
                dangerouslySetInnerHTML={{ __html: result.value }}
                className={classes.line}
              ></li>
            );
          })}
        </ul>
        <div className="logs-anthor" ref={elementRef}></div>
      </>
    );
  };

  useEffect(() => {
    if (elementRef.current) {
      elementRef.current.scrollIntoView({ behavior: "auto" });
    }
  }, [logs]);

  useEffect(() => {
    const query: qs.ParsedQs = qs.parse(search, {
      ignoreQueryPrefix: true,
    });
    fetchLogs({
      logId: query.logId as string,
      pod: query.pod as string,
      container: query.container as string,
      kubeConfig: query.kubeConfig as string,
      tail: LOG_TAIL_COUNT,
    });
    const timer = setInterval(() => {
      fetchLogs({
        logId: query.logId as string,
        pod: query.pod as string,
        container: query.container as string,
        kubeConfig: query.kubeConfig as string,
        tail: LOG_TAIL_COUNT,
      });
    }, LOG_INTERVAL_MS);
    return () => {
      clearInterval(timer);
    };
  }, [search]);

  useEffect(() => {
    const $link = document.getElementById("syntax-theme");
    if (!$link) {
      return;
    }
    const { light, dark } = $link.dataset;
    if (theme === ThemeType.light) {
      $link.setAttribute("href", light);
    } else {
      $link.setAttribute("href", dark);
    }
  }, [theme]);

  return (
    <div className={classes.root}>
      {logs.items.length > 0 ? renderList() : renderNoContent()}
    </div>
  );
};

export default Logs;
