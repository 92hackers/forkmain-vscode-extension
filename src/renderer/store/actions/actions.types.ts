import { IDeployments, ILogs } from "../store.types";

export enum ActionType {
  redirect = "redirect",
  toggleTheme = "theme/toggle",
  updateLogs = "logs/update",
  updateDeployments = "deployments/update",
}

export interface IRedirect {
  type: ActionType.redirect;
  payload: { url: string };
}

export interface IToggleTheme {
  type: ActionType.toggleTheme;
  payload: { theme: string };
}

export interface IUpdateLogs {
  type: ActionType.updateLogs;
  payload: { logs: ILogs };
}

export interface IUpdateDeployments {
  type: ActionType.updateDeployments;
  payload: { deployments: IDeployments };
}

export type Actions =
  | IRedirect
  | IToggleTheme
  | IUpdateLogs
  | IUpdateDeployments;
