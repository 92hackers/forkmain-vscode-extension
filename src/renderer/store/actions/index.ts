import { ActionType, IRedirect } from "./actions.types";

export const redirect = (uri: string): IRedirect => ({
  type: ActionType.redirect,
  payload: { uri },
});
