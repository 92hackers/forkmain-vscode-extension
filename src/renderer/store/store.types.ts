import { Actions } from "./actions/actions.types";

export interface IStoreState {
  url: string;
  theme: string;
}

export interface IAppContext {
  state: IStoreState;
  dispatch: React.Dispatch<Actions>;
}
