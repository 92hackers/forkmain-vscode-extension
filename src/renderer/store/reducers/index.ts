import { IStoreState } from "../store.types";
import { Actions, ActionType, IRedirect } from "../actions/actions.types";
import { redirect } from "./redirect";

const reducer = (state: IStoreState, action: Actions) => {
  switch (action.type) {
    case ActionType.Redirect:
      return redirect(state, action as IRedirect);
    default:
      return state;
  }
};

export default reducer;
