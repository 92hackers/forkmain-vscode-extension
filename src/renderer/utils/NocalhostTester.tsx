import { mount, ReactWrapper } from "enzyme";
import React, {
  MutableRefObject,
  useContext,
  useEffect,
  useRef,
  useState,
} from "react";
import { AppWrapper } from "../App";
import AppRouter from "../AppRouter";
import { Actions } from "../store/actions/actions.types";
import { store } from "../store/store";

type onMountHandler = (dispatch: React.Dispatch<Actions>) => void;
type TestRunner = () => void;

interface TesterProps {
  onMount: onMountHandler;
}

const Tester = (props: TesterProps): JSX.Element => {
  const { dispatch } = useContext(store);
  const [renderedId, setRenderedId] = useState(1);
  const elementRef: MutableRefObject<HTMLDivElement> = useRef<HTMLDivElement>(
    null
  );
  const { onMount } = props;

  const observe = (element: HTMLElement): MutationObserver => {
    const observer = new MutationObserver((mutations: MutationRecord[]) => {
      mutations.forEach((record: MutationRecord) => {
        const runner: TestRunner | null = NocalhostTester.getRunner();
        const delay: number = NocalhostTester.getDelay();
        if (runner) {
          setTimeout(runner, delay);
        }
      });
    });
    observer.observe(elementRef.current, {
      attributes: true,
      attributeFilter: ["data-rendered-id"],
    });
    return observer;
  };

  useEffect(() => {
    const observer: MutationObserver = observe(elementRef.current);
    onMount(dispatch);
    setRenderedId(renderedId + 1);
    () => {
      observer.disconnect();
    };
  }, []);

  return <div ref={elementRef} data-rendered-id={renderedId}></div>;
};

export default class NocalhostTester {
  private wrapper: ReactWrapper | null = null;
  private static testRunner: TestRunner | null = null;
  private static delay: number = 30;

  public static getRunner(): TestRunner | null {
    return NocalhostTester.testRunner;
  }

  public static getDelay(): number {
    return NocalhostTester.delay;
  }

  constructor(onMount: onMountHandler) {
    this.wrapper = mount(
      <AppWrapper>
        <>
          <AppRouter />
          <Tester onMount={onMount} />
        </>
      </AppWrapper>
    );
  }

  getWrapper(): ReactWrapper | null {
    return this.wrapper;
  }

  run(runner: TestRunner, delay = 30): void {
    NocalhostTester.testRunner = runner;
    NocalhostTester.delay = delay;
  }
}
