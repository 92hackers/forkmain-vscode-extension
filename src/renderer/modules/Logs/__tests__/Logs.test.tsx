import React from "react";
import { ReactWrapper } from "enzyme";
import Logs from "../index";
import Landing from "../../Landing";
import NocalhostTester from "../../../utils/NocalhostTester";
import { Actions } from "../../../store/actions/actions.types";
import { redirect } from "../../../store/actions";

describe("Test <Logs />", () => {
  it("should display no content", (done) => {
    const tester: NocalhostTester = new NocalhostTester(
      (dispatch: React.Dispatch<Actions>) => {
        dispatch(redirect("/logs"));
      }
    );

    tester.run(() => {
      const wrapper: ReactWrapper | null = tester.getWrapper();
      console.log(wrapper.html());
      expect(wrapper.find(Logs)).toHaveLength(1);
      expect(wrapper.find(Landing)).toHaveLength(0);
      done();
    });
  });
});
