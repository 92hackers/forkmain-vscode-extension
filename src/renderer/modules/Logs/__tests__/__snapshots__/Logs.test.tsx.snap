// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Logs /> should display no content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-44"
      >
        <div
          class="makeStyles-root-45"
          data-testid="logs"
        >
          <div
            class="makeStyles-noContent-46"
          >
            No Content.
          </div>
        </div>
      </div>
      <div
        data-rendered-id="2"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-44"
    >
      <div
        class="makeStyles-root-45"
        data-testid="logs"
      >
        <div
          class="makeStyles-noContent-46"
        >
          No Content.
        </div>
      </div>
    </div>
    <div
      data-rendered-id="2"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test <Logs /> should disploy logs 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-72"
      >
        <div
          class="makeStyles-root-73"
          data-testid="logs"
        >
          <ul
            class="makeStyles-list-75"
          >
            <li
              class="makeStyles-line-76"
              data-line="1"
            >
              <span
                class="hljs-type"
              >
                AAA
              </span>
            </li>
            <li
              class="makeStyles-line-76"
              data-line="2"
            >
              <span
                class="hljs-type"
              >
                BBB
              </span>
            </li>
            <li
              class="makeStyles-line-76"
              data-line="3"
            >
              <span
                class="hljs-type"
              >
                CCC
              </span>
            </li>
          </ul>
          <div
            class="logs-anthor"
          />
        </div>
      </div>
      <div
        data-rendered-id="2"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-72"
    >
      <div
        class="makeStyles-root-73"
        data-testid="logs"
      >
        <ul
          class="makeStyles-list-75"
        >
          <li
            class="makeStyles-line-76"
            data-line="1"
          >
            <span
              class="hljs-type"
            >
              AAA
            </span>
          </li>
          <li
            class="makeStyles-line-76"
            data-line="2"
          >
            <span
              class="hljs-type"
            >
              BBB
            </span>
          </li>
          <li
            class="makeStyles-line-76"
            data-line="3"
          >
            <span
              class="hljs-type"
            >
              CCC
            </span>
          </li>
        </ul>
        <div
          class="logs-anthor"
        />
      </div>
    </div>
    <div
      data-rendered-id="2"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test <Logs /> should render skeleton 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-16"
      >
        <div
          class="makeStyles-root-17"
          data-testid="logs"
        >
          <div
            class="makeStyles-skeletonContainer-21"
          >
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
            <div
              class="makeStyles-skeletonItem-22"
              id="logs-skeleton"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        data-rendered-id="2"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-16"
    >
      <div
        class="makeStyles-root-17"
        data-testid="logs"
      >
        <div
          class="makeStyles-skeletonContainer-21"
        >
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
          <div
            class="makeStyles-skeletonItem-22"
            id="logs-skeleton"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      data-rendered-id="2"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
